sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/AllBranchesShouldNotHaveSameImplementation.cs

```diff
@@ -100,7 +100,8 @@ protected override void Initialize(SonarAnalysisContext context)
                 {
                     var ternaryStatement = (ConditionalExpressionSyntax)c.Node;
 
-                    if (ternaryStatement.WhenTrue.IsEquivalentTo(ternaryStatement.WhenFalse, false))
+                    if (ternaryStatement.WhenTrue.RemoveParentheses().IsEquivalentTo(
+                            ternaryStatement.WhenFalse.RemoveParentheses(), false))
```

Comment:
Is this related to the current change set?

Reply:
Yes when copying the test cases from the deprecated rules I have noticed we were not handling parenthesis (on the new rule) while the deprecated one was.